SalesforceCommerceCloud/pwa-kit
# @W-16567627 Add useDNT hook to commerce-sdk-react

Add the hook that allows merchant to update DNT preference and gives a boolean on whether or not the DNT preference exists. This will be consumed by template retail react app.

Plan for updateDNT usage:
1. If the user gives a Yes or No to consent tracking form, updateDNT(true | false) will be called. This will create `dw_dnt` cookie with expiration set to match the refresh token.
2. If the user gives no clear answer and closes the consent tracking form, updateDNT(null) will be called. This will create `dw_dnt` set to expire when browser is closed. The value of the cookie will either be `defaultDnt` passed into CommerceApiProvider or the SLAS default value.

Flow:
![Screenshot 2024-09-06 at 1 10 03 PM](https://github.com/user-attachments/assets/b1a38f9d-9957-4e56-a6d8-a44b9bef1369)
https://lucid.app/lucidchart/ee695d9a-b0b2-43b9-bfbd-c9a0c4b3196f/edit?invitationId=inv_84e0f60d-9ca5-4d34-b852-d1baaef20471&page=3gd-sRgvdGID#

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1) Add useDNT hook
- (change1) Refactored auth/index to take the refresh token flow out, since it is needed for updateDNT
- (change2) Updated test-commerce-sdk-react

# How to Test-Drive This PR

- (step1) `npm start` in test-commerce-sdk-react, go to the DNT page
- (step2) Open developer tools. Go to Application and see that there is no `dw_dnt`. If there is, delete it.
- (step3) Click on the blue button. See that `dw_dnt` is created and the blue button is replaced by a confirmation text saying `DNT cookie is successfully set`
- (step4) Close the browser, and reopen the page to see that the cookie was deleted.
- (step5) Ensure tests pass on `commerce-sdk-react`

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
